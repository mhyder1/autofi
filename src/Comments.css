.comments {
    margin-left: 20px;
    border-bottom: 1px solid black;
}

.comments span {
    font-weight: bold;
}