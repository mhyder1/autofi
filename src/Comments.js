import React, { useState, useEffect } from "react";
import "./Comments.css";
import Form from './Form'
export default function Comments({ id }) {

  const [comments, setComments] = useState([]);
  

  const getComments = async (id) => {
    const res = await fetch(
      `https://jsonplaceholder.typicode.com/comments?postId=${id}`
    );
    const comments = await res.json();
    setComments(comments);
  };

  useEffect(() => {
    getComments(id);
  }, [id]);

  return (
      <div>
        {comments.map(({ id, name, email, body }) => (
            <section key={id} className="comments">
            <p>
                <span>Name:</span> {name}
            </p>
            <p>
                <span>Email:</span> {email}
            </p>
            <p>
                <span>Body:</span> {body}
            </p>
            </section>
        ))}
        <Form comments={comments} setComments={setComments} />
    </div>
  )
}