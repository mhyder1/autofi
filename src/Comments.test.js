import React from 'react';
import ReactDOM from 'react-dom';
import Comments from './Comments';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme';

describe(`Comments Component`, () => {    
    it('Smoke test',() =>{
        const id = 1
        const div = document.createElement('div');
        ReactDOM.render(<Comments id={id} />,div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it("Returns empty list item, if nothing passed.",() => {
        const id = 1
        const wrapper = shallow(<Comments id={id} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})