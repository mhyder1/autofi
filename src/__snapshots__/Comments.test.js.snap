// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments Component Returns empty list item, if nothing passed. 1`] = `null`;
